grails-app/controllers/org/broadinstitute/orsp/api/SampleConsentLinkController.groovy

```diff
@@ -0,0 +1,44 @@
+package org.broadinstitute.orsp.api
+
+import com.google.gson.JsonParser
+import grails.converters.JSON
+import grails.rest.Resource
+import groovy.util.logging.Slf4j
+import org.broadinstitute.orsp.AuthenticatedController
+import org.broadinstitute.orsp.ConsentCollectionLink
+import org.broadinstitute.orsp.User
+import org.broadinstitute.orsp.utils.IssueUtils
+import org.springframework.web.multipart.MultipartFile
+
+@Slf4j
+@Resource(readOnly = false, formats = ['JSON', 'APPLICATION-MULTIPART'])
+class SampleConsentLinkController extends AuthenticatedController {
+
+    def pages() {
+        render(view: "/linkWizard/index")
+    }
+
+    def save() {
+        JsonParser parser = new JsonParser()
+        User user = getUser()
+        List<MultipartFile> files = request.multiFileMap.collect { it.value }.flatten()
+        ConsentCollectionLink consentCollectionLink = IssueUtils.getJson(ConsentCollectionLink.class, parser.parse(request.parameterMap["dataConsentCollection"].toString())[0])
+
+        consentCollectionLink.creationDate = new Date()
+        try {
+            persistenceService.saveConsentCollectionLink(consentCollectionLink)
+            if (!files?.isEmpty()) {
+                files.forEach {
+                    storageProviderService.saveMultipartFile(user.displayName, user.userName, null, it.name, it, consentCollectionLink)
+                }
+            }
+
+            response.status = 201
+            render([message: consentCollectionLink] as JSON)
+        } catch (Exception e) {
+            flash.error = e.getMessage()
+            response.status = 500
+            render([error: e.getMessage()] as JSON)
```

Comment:
If you hit an error partway through a list of files being uploaded, there will be `N` files in an invalid state and a partially saved `consentCollectionLink`. I think this could use some better cleanup handling.

Reply:
Sure, good idea.